### ISSUE TITLE
MKD MKA KONQUEST ANIM - MKD KONQUEST MAP

### ISSUE THREAD

**Issue by 972oTeV** [leeao/MortalKombat#4]
Hi again 😅
i was extracting some models from MKD, 
and i've checked the animations from Konquest mode and:
![MKD konquest Anim](https://github.com/leeao/MortalKombat/assets/122237320/b801b57a-107a-4752-8922-aacac1fb9b6c)

And like always, here are some samples to test: 
https://mega.nz/file/Q0ZH1Kjb#edzubuZRlxmeRfm2abB8pOfPZ0cBRUWfTs0J4dOZbWM 

PD: Some animations from MKA Konquest mode (and some fighters too), also have problems in arms, like this one:
![Screenshot](https://github.com/leeao/MortalKombat/assets/122237320/5c97f383-c9ca-4ab0-96f1-eef5c8989ea4)
Samples here:
https://mega.nz/file/Ig5SwIgA#uowR-ZR21u3gFXp9RFc_wxRNJTS_N7ja2Ro6-hse4vQ

PD 2: Some animations from MKA Fighters looks like this: (also Jade, Noob, and 4 Arms Chars, the extra arms still in T-pose)
![mokap](https://github.com/user-attachments/assets/5e93cbd1-ae6f-430c-9bbd-ab4e652c8c61)
Sample Here:
https://mega.nz/file/Q4gHQIKD#oudjOO9C0UzlBFQ12WrdodrEmJZvk9A9QigqjBtL2Ig 

PD 3:
The maps from Konquest comes out in 2 pieces, ground and objects, but all the objects comes with bones but in the same point, because doesn't have Vertex groups, there are some solution for this problem? 🤔 
![Screenshot_5](https://github.com/user-attachments/assets/0c903f7e-af15-4c59-b6ba-7acc7a10a149)

Samples:
https://mega.nz/file/dsQghS6Y#gDSBxoLTeCSHqJs4IouLnkA3815sHd-e0c8BRgVyKFA

Thanks! 😃

**Comment by HunterTheReckoning** [comment]
I looked into this myself. Many of the MKD Konquest stage object information is stored in 3.dat. Running a parser for MAP/SEC sections revealed they have their own armature which doesn't share the MKSkinBones that the plugin seeks. It may require more research or a separate plugin strictly for Konquest stuff, Which explains why the OBJECTS.dff don't go into the positions of the bones and why TILES.dff exports without any.

In short: Tiles.dff isn't importing with an armature, and Objects.dff doesn't mount to its armature. Object Information on them is stored in 3.dat under the kon_..._1.ssf

**Comment by 972oTeV** [comment]
Hi!
Thanks for the reply, i thought no one will answer this, but the only way i found to extract the maps, it's this one:

convert the Objects.dff in a .obj model,
later open it with blender, then re-save all the parts separately,
and after that, reorganized all meshes in .obj models
But i don't want to put every part in the map, that will be the Hell.

If you want or know how to modify the script, go for it, i can handle the models like i said, and if i share them, someone else can make the full map, not me.

> I looked into this myself. Many of the MKD Konquest stage object information is stored in 3.dat. Running a parser for MAP/SEC sections revealed they have their own armature which doesn't share the MKSkinBones that the plugin seeks. It may require more research or a separate plugin strictly for Konquest stuff, Which explains why the OBJECTS.dff don't go into the positions of the bones and why TILES.dff exports without any.
> 
> In short: Tiles.dff isn't importing with an armature, and Objects.dff doesn't mount to its armature. Object Information on them is stored in 3.dat under the kon_..._1.ssf

**Comment by HunterTheReckoning** [comment]
<img width="1735" height="705" alt="Image" src="https://github.com/user-attachments/assets/b153d2a1-ec6c-4253-937a-74aebd443fd2" />

<img width="1410" height="778" alt="Image" src="https://github.com/user-attachments/assets/142b6d2f-01f4-4633-be25-77d6ee4eb298" />

I wrote a blender script and updated the plugin on my end to get the original names for the meshes. The blender script then groups them together by the name they have.
The names go as bone#_split#_part#. This makes it easier to sort them out. The object hierarchy is set by bones. Tiles.dff has bones but the plugin doesn't seem to rip them right, nor the vertex groups to put them in place. Do if you want to manually do it, the script in blender and update for actual mesh names helps a lot.

**Comment by 972oTeV** [comment]
> <img alt="Image" width="1735" height="705" src="https://private-user-images.githubusercontent.com/104789763/584969766-b153d2a1-ec6c-4253-937a-74aebd443fd2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Nzc1MTM3MzgsIm5iZiI6MTc3NzUxMzQzOCwicGF0aCI6Ii8xMDQ3ODk3NjMvNTg0OTY5NzY2LWIxNTNkMmExLWVjNmMtNDI1My05MzdhLTc0YWViZDQ0M2ZkMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQzMFQwMTQzNThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMWIzMjAwYTU2MDcwYjU0NTViM2JmN2E5YjczN2Y1YzIzNDZkMzVmODE1N2I2NGIwNDA3ZmY3YjVkNWY5N2UyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.Kz3ZwQ7uV4zW89m0Se1jN11TDNs2zTR57dsidRzvAI4"> <img alt="Image" width="1410" height="778" src="https://private-user-images.githubusercontent.com/104789763/584969823-142b6d2f-01f4-4633-be25-77d6ee4eb298.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Nzc1MTM3MzgsIm5iZiI6MTc3NzUxMzQzOCwicGF0aCI6Ii8xMDQ3ODk3NjMvNTg0OTY5ODIzLTE0MmI2ZDJmLTAxZjQtNDYzMy1iZTI1LTc3ZDZlZTRlYjI5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQzMFQwMTQzNThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04OTE4ODI5NjlkMDc4YWRkNDEyMDZlNjljNTIxMWJiYzFkYTFiMzAzNDljZjQ0M2JjMGQzYjVhOWNiMmJhYTFjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.IRMcCVVnOZn1XfZOgn_xB-DwVzCHSjQ9yL-5Ywb0fd4">
> I wrote a blender script and updated the plugin on my end to get the original names for the meshes. The blender script then groups them together by the name they have. The names go as bone#_split#_part#. This makes it easier to sort them out. The object hierarchy is set by bones. Tiles.dff has bones but the plugin doesn't seem to rip them right, nor the vertex groups to put them in place. Do if you want to manually do it, the script in blender and update for actual mesh names helps a lot.

Ooooh nice! it's good to have some help with that, can i download that update or the script for blender? i wanna try :)

**Comment by 972oTeV** [comment]
> <img alt="Image" width="1735" height="705" src="https://private-user-images.githubusercontent.com/104789763/584969766-b153d2a1-ec6c-4253-937a-74aebd443fd2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Nzg4ODU2NzcsIm5iZiI6MTc3ODg4NTM3NywicGF0aCI6Ii8xMDQ3ODk3NjMvNTg0OTY5NzY2LWIxNTNkMmExLWVjNmMtNDI1My05MzdhLTc0YWViZDQ0M2ZkMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUxNVQyMjQ5MzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNDE4OWYxNTM1NjYxN2RiMzVjOTg0ZmM4Zjg1ZTgwNDc2YmViMWQ5YTFkZTY0OWY1NzU1YmM0ZjdlZjY3NjgyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.XrYiXyvu8Y9qq4CbXlLWUfDb-u-IopY5UeHlD8uQOnQ"> <img alt="Image" width="1410" height="778" src="https://private-user-images.githubusercontent.com/104789763/584969823-142b6d2f-01f4-4633-be25-77d6ee4eb298.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Nzg4ODU2NzcsIm5iZiI6MTc3ODg4NTM3NywicGF0aCI6Ii8xMDQ3ODk3NjMvNTg0OTY5ODIzLTE0MmI2ZDJmLTAxZjQtNDYzMy1iZTI1LTc3ZDZlZTRlYjI5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUxNVQyMjQ5MzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMGUwODhjODM1NGU3MTMxOTJiZjVmNjVkZWY2ZWJhOWU0NTc1MzhkMDEzOWQzYmMzNjRhOGE1ZjZmOWIwYTFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.QiNitZIC0wQ8BYr_MnH8ncdOuQ3soFey5DGP23h5MeI">
> I wrote a blender script and updated the plugin on my end to get the original names for the meshes. The blender script then groups them together by the name they have. The names go as bone#_split#_part#. This makes it easier to sort them out. The object hierarchy is set by bones. Tiles.dff has bones but the plugin doesn't seem to rip them right, nor the vertex groups to put them in place. Do if you want to manually do it, the script in blender and update for actual mesh names helps a lot.

Hello?

**Comment by HunterTheReckoning** [comment]
> > <img alt="Image" width="1735" height="705" src="https://private-user-images.githubusercontent.com/104789763/584969766-b153d2a1-ec6c-4253-937a-74aebd443fd2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Nzg4ODU2NzcsIm5iZiI6MTc3ODg4NTM3NywicGF0aCI6Ii8xMDQ3ODk3NjMvNTg0OTY5NzY2LWIxNTNkMmExLWVjNmMtNDI1My05MzdhLTc0YWViZDQ0M2ZkMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUxNVQyMjQ5MzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNDE4OWYxNTM1NjYxN2RiMzVjOTg0ZmM4Zjg1ZTgwNDc2YmViMWQ5YTFkZTY0OWY1NzU1YmM0ZjdlZjY3NjgyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.XrYiXyvu8Y9qq4CbXlLWUfDb-u-IopY5UeHlD8uQOnQ"> <img alt="Image" width="1410" height="778" src="https://private-user-images.githubusercontent.com/104789763/584969823-142b6d2f-01f4-4633-be25-77d6ee4eb298.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Nzg4ODU2NzcsIm5iZiI6MTc3ODg4NTM3NywicGF0aCI6Ii8xMDQ3ODk3NjMvNTg0OTY5ODIzLTE0MmI2ZDJmLTAxZjQtNDYzMy1iZTI1LTc3ZDZlZTRlYjI5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUxNVQyMjQ5MzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMGUwODhjODM1NGU3MTMxOTJiZjVmNjVkZWY2ZWJhOWU0NTc1MzhkMDEzOWQzYmMzNjRhOGE1ZjZmOWIwYTFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.QiNitZIC0wQ8BYr_MnH8ncdOuQ3soFey5DGP23h5MeI">
> > I wrote a blender script and updated the plugin on my end to get the original names for the meshes. The blender script then groups them together by the name they have. The names go as bone#_split#_part#. This makes it easier to sort them out. The object hierarchy is set by bones. Tiles.dff has bones but the plugin doesn't seem to rip them right, nor the vertex groups to put them in place. Do if you want to manually do it, the script in blender and update for actual mesh names helps a lot.
> 
> Hello?

https://docs.google.com/document/d/1qEI9ZsZyoMAEF6ahf3tPRSYECiF-0KodBOEpsIO7gXI/edit?usp=drivesdk

**Comment by 972oTeV** [comment]
> > Hello?
> 
> https://docs.google.com/document/d/1qEI9ZsZyoMAEF6ahf3tPRSYECiF-0KodBOEpsIO7gXI/edit?usp=drivesdk

Thanks a lot, i'll test them later :D 
Edit: Works really good, Thanks again, and i asked gemini to make another script to convert all the groups in separated Obj files, and now the process it's even easy :D (Some normals are flipped but well, details)
